query/config.go

```diff
@@ -4,15 +4,264 @@ import (
+	Filter   map[string]string            `json:"filter"` // Filter by labels
+	GroupBy  []string                     `json:"groupBy"`
+	Tags     []string                     `json:"tags"`
+}
+
+func (t *ConfigSummaryRequest) OrderBy() string {
+	var output []string
+	for i := 0; i < len(t.GroupBy); i++ {
+		output = append(output, fmt.Sprintf("%d", i+1))
+	}
+	return strings.Join(output, ", ")
+}
+
+func (t *ConfigSummaryRequest) analysisJoin() string {
+	output := "LEFT JOIN aggregated_analysis_count ON "
+	var clauses []string
+	for _, g := range t.GroupBy {
+		switch g {
+		case "type":
+			clauses = append(clauses, "aggregated_analysis_count .type = config_items.type")
+		default:
+			clauses = append(clauses, fmt.Sprintf("aggregated_analysis_count.%s = config_items.tags->>'%s'", g, g))
+		}
+	}
+
+	return output + strings.Join(clauses, " AND ")
+}
+
+func (t *ConfigSummaryRequest) changesJoin() string {
+	output := "LEFT JOIN changes_grouped ON "
+	var clauses []string
+	for _, g := range t.GroupBy {
+		switch g {
+		case "type":
+			clauses = append(clauses, "changes_grouped.type = config_items.type")
+		default:
+			clauses = append(clauses, fmt.Sprintf("changes_grouped.%s = config_items.tags->>'%s'", g, g))
+		}
+	}
+
+	return output + strings.Join(clauses, " AND ")
+}
+
+func (t *ConfigSummaryRequest) plainSelectClause(appendSelect ...string) []string {
+	return append(t.GroupBy, appendSelect...)
+}
+
+func (t *ConfigSummaryRequest) summarySelectClause() []string {
+	cols := []string{
+		fmt.Sprintf("SUM(cost_total_%s) as cost_%s", t.Cost, t.Cost),
+		"changes_grouped.count AS changes",
+		"aggregated_analysis_count.analysis AS analysis",
+		"COUNT(*) AS total_configs",
+	}
+
+	return t.baseSelectClause(cols...)
+}
+
+func (t *ConfigSummaryRequest) baseSelectClause(appendSelect ...string) []string {
+	var output []string
+	for _, g := range t.GroupBy {
+		switch g {
+		case "type":
+			output = append(output, "config_items.type")
+		default:
+			output = append(output, fmt.Sprintf("config_items.tags->>'%s' as %s", g, g))
+		}
+	}
+
+	for _, tag := range t.Tags {
+		output = append(output, fmt.Sprintf("config_items.tags->>'%s' as %s", tag, tag))
+	}
+
+	if len(output) == 0 {
+		output = []string{"config_items.type"}
+	}
+
+	output = append(output, appendSelect...)
+	return output
+}
+
+func (t *ConfigSummaryRequest) groupBy() []string {
+	var output []string
+	for _, g := range t.GroupBy {
+		switch g {
+		case "type":
+			output = append(output, "config_items.type")
+		default:
+			output = append(output, fmt.Sprintf("config_items.tags->>'%s'", g))
+		}
+	}
+
+	for _, tag := range t.Tags {
+		output = append(output, fmt.Sprintf("config_items.tags->>'%s'", tag))
+	}
+
+	return output
+}
+
+func (t *ConfigSummaryRequest) SetDefaults() {
+	if t.Changes.Since == "" {
+		t.Changes.Since = "2d"
+	}
+
+	if t.Analysis.Since == "" {
+		t.Analysis.Since = "2d"
+	}
+
+	if t.Cost == "" {
+		t.Cost = "30d"
+	}
```

Comment:
I don't think we should default these, If not specified we don't run those queries

Can we also add a health summary (healthy, unhealthy, unknown, warning, pending (!ready)